helm/cluster-api-core/templates/_helpers.tpl

```diff
@@ -27,4 +27,5 @@ Selector labels
 {{- define "labels.selector" -}}
 app.kubernetes.io/name: {{ include "name" . | quote }}
 app.kubernetes.io/instance: {{ .Release.Name | quote }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
```

Comment:
We need to add the version to the pod selectors so we don't select other versions.

Reply:
I think you need to remove it from the labels in that case